osprey/processes/wps_convolution.py

```diff
@@ -77,33 +129,33 @@ def _handler(self, request, response):
         log_handler(
             self, response, "Starting Process", process_step="start", level=loglevel
         )
-        config = request.inputs["config"][0].data
-
-        if version == "1.1.1":
-            if os.path.isfile(config):
-                config_dict = read_config(config)
-            else:
-                config_dict = json.loads(config)
-
-            log_handler(
-                self,
-                response,
-                "Run Flux Convolution",
-                process_step="process",
-                level=loglevel,
-            )
-            convolution(config_dict)
-
-        elif version == "1.1.0-1":
-            config_dict = read_config(config)
-            log_handler(
-                self,
-                response,
-                "Run Flux Convolution",
-                process_step="process",
-                level=loglevel,
-            )
-            convolution(config)
+        unprocessed = request.inputs["config"][0].data
+
+        log_handler(
+            self,
+            response,
+            "Run Flux Convolution",
+            process_step="process",
+            level=loglevel,
+        )
+
+        if os.path.isfile(unprocessed):
+            config = read_config(unprocessed)
+            if version == "1.1.0-1":  # RVIC1.1.0.post1
+                cfg_filepath = unprocessed
+                convolution(cfg_filepath)
+            elif version == "1.1.1":  # RVIC1.1.1
+                convolution(config)
+        else:
+            unprocessed = unprocessed.replace("'", '"')
+            config = config_hander(self.workdir, unprocessed, self.config_template)
+            if version == "1.1.0-1":  # RVIC1.1.0.post1
+                cfg_filepath = config_file_builder(
+                    self.workdir, config, self.config_template
+                )
+                convolution(cfg_filepath)
+            elif version == "1.1.1":  # RVIC1.1.1
+                convolution(config)
```

Comment:
This pattern looks very similar to the one above. There may be room to factor this out a bit.